src/components/popover/input_popover.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+
+import { EuiFocusTrap } from '../focus_trap';
+import { EuiPopover, EuiPopoverPropTypes } from './popover';
+
+export const EuiInputPopover = ({
```

Comment:
I know we mostly just use this with inputs, but really all this component does is "attach" the popover to the triggering element and match the width. Perhaps we can make this more generic, mainly in the concern of naming and call it `EuiPopoverAttached`. Then it would probably change the prop `input` to `anchor`.

Reply:
Oh I see now, there **is** the option to "attach" the normal EuiPopover, but this new component ensures it matches the width...

Reply:
And traps focus

Reply:
To be honest, I'm not sure right now.
As is, I'm running into difficulties capturing the necessary key press and focus events because of limited access to the underlying form components.

So as a general `EuiPopoverAttached` component this works well. But when coupled with having to discern events from form controls, it needs a bit more to accomplish a11y needs.

Reply:
Would it then, at least to get this style need in, just be easier to allow EuiPopover's `attachToAnchor` prop also accept a string like `attachToAnchor="matchWidth"` or something? So then this component could just be dedicated to all that a11y stuff and isn't necessary **right now**.

Reply:
I think I still prefer the move to a new component, rather than rely on configuration. The a11y problem doesn't _really_ get any easier just using `EuiPopover` with an extra prop, anyway.

For the range case in particular, I think the current state of things is valid:
1) click opens the popover for more clarity, and tab will eventually close it.
2) keyboard nav in does not open the popover, but a keyboard savvy user still gets the affordance of a `number` input.

Thoughts?

Reply:
All that to say, we'll see changes as future form controls want to uses this pattern. But a lot of that will be influenced by the broader popover discussion that's ongoing.

Reply:
Yeah those make sense to me

Reply:
Are you happy with where this is at currently then? I can merge it in if so.

Reply:
Yeah we can at least let others add opinions in the main thread